### PR TITLE
Feature/dplah 67 harvest all from intermediate provider

### LINKED ISSUE #67: Ability to harvest all or delete all from one Intermediate Provider
"Button to harvest all or delete all from one Intermediate Provider (e.g. POWER Library)
-- What is the best way to configure this?"

### PR DESCRIPTION
**Pull request by kelynch**
fixes #67

### CODE REVIEW

File: app/views/providers/index.html.erb [hunk 1]
```diff
@@ -37,7 +37,8 @@
...
-        <td class="oai-actions btn-provider"><%= link_to 'Harvest all OAI Seeds from this Institution', harvest_all_by_institution_provider_path(provider), method: :post, class: :triggerWaiting, data: { confirm: 'This may take a while for institutions with many OAI seeds. Are you sure?' } %></td>
+        <td class="oai-actions btn-provider"><%= link_to 'Harvest all OAI Seeds from this Institution', harvest_all_selective_contributing_institution_provider_path(provider), method: :post, class: :triggerWaiting, data: { confirm: 'This may take a while for institutions with many OAI seeds. Are you sure?' } %></td>
+        <td class="oai-actions btn-provider"><%= link_to 'Harvest all OAI Seeds from this Institution\'s Intermediate Provider', harvest_all_selective_intermediate_provider_provider_path(provider), method: :post, class: :triggerWaiting, data: { confirm: 'This may take a while for intermediate providers with institutions and associated OAI seeds. Are you sure?' } %></td>
```
Comment (bibliotechy) [inline app/views/providers/index.html.erb]: @kelynch Is it worth disabling this button for providers that don't have an intermediate provider?

Comment (kelynch) [inline app/views/providers/index.html.erb]: Sounds good!  I added a method for this.  Could be repurposed in this view wholesale if we add more attribute-dependent actions later on?

**Comment by bibliotechy** [comment]
Thanks @kelynch

**Comment by bibliotechy** [comment]
One note I want to leave here, maybe for future development. Maybe we turn this into another issue depending on metadata team feedback. 

Because intermediate provider triggers all harvests as a single job, if one of the seed's harvest fails, you have to run all of the already successful ones again in order to continue. It might be nice to have the harvest all from intermediate create a regular harvest job for each seed, so that if one fails, that can be dealt with individually, instead of having to rerun all seeds in the intermediate provider.

I'd suggest we wait to hear feedback from metadata team about how they work before we think about doing this.

**Comment by kelynch** [comment]
👍  I like this suggestion.  Would be a good use of future development time.

**Comment by bibliotechy** [comment]
@kelynch @EliseTemple This merged OK in my local repo, but when I merged into aggtest and onto the aggregator-testing, it caused the rails app to 500  with: 
```
ActionView::Template::Error (undefined method `last_harvested' for #<Provider:0x007fcf626e3ac8>):
    34:         <td><%= auto_link(provider.endpoint_url) %></td>
    35:         <td><%= provider.set %></td>
    36:         <td><%= provider.collection_name %></td>
    37:         <td><%= human_readable_time(provider.last_harvested) %></td>
    38:         <td><%= provider.contributing_institution %></td>
    39:         <td><%= provider.intermediate_provider %></td>
    40:         <td><%= provider.common_repository_type %></td>
  app/views/providers/index.html.erb:37:in `block in _app_views_providers_index_html_erb__324047024881704893_70264342868480'
  app/views/providers/index.html.erb:31:in `_app_views_providers_index_html_erb__324047024881704893_70264342868480'
``` 

I'm reverting this merge so we can deploy other recently added features while I investigate.

**Comment by kelynch** [comment]
Was the database on aggtest updated?  Last_harvested is a new field; the migration is in the pull request.

**Comment by bibliotechy** [comment]
@kelynch I thought there was a migration too, but running `db:migrate` did not create an update.
 
I went back and looked at the code and it looks like that migration was removed after it was added. 
https://github.com/tulibraries/dplah/pull/103/files#diff-ab26e82244702c4d5f2c32373cfb0bfaL1

and 

https://github.com/tulibraries/dplah/pull/103/files#diff-1acd2e7e27a227829d5d14a91c863bb6L64

Removed by commit https://github.com/tulibraries/dplah/pull/103/commits/fdfe23e5ca5fc9070fb57bf1b7d81cbf28ea3cf5

**Comment by kelynch** [comment]
I see what happened.  This PR was submitted against develop before #72 was merged (after it was submitted).  Guh, this branch is kind of in bad shape.  I'm going to rebase and submit a new PR with the migration back in place.

**Comment by dorevabelfiore-temple** [comment]
Anastasia will be testing the Harvest All from Institution feature week of 6/27 post ALA Annual.

**Pull request merged**